Git-commit: b79399510888998778ea6a3a281e30cbe59fdb37
From: Joe Thornber <ejt@redhat.com>
Date: Fri, 27 Jul 2012 15:08:13 +0100
Subject: [PATCH] dm thin metadata: remove duplicate pmd initialisation
Patch-mainline: v3.6-rc1
Reference: FATE#313903

Remove some duplicate initialisation of struct dm_pool_metadata.

These pmd fields are initialised by both:
  __format_metadata's calls to dm_btree_empty
  __write_initial_superblock + __begin_transaction

Signed-off-by: Joe Thornber <ejt@redhat.com>
Signed-off-by: Mike Snitzer <snitzer@redhat.com>
Signed-off-by: Alasdair G Kergon <agk@redhat.com>
Acked-by: NeilBrown <nfbrown@suse.com>

---
 drivers/md/dm-thin-metadata.c |    5 -----
 1 file changed, 5 deletions(-)

--- linux-3.0-SLE11-SP3.orig/drivers/md/dm-thin-metadata.c
+++ linux-3.0-SLE11-SP3/drivers/md/dm-thin-metadata.c
@@ -512,11 +512,6 @@ static int __format_metadata(struct dm_p
 
 	__setup_btree_details(pmd);
 
-	pmd->root = 0;
-	pmd->details_root = 0;
-	pmd->trans_id = 0;
-	pmd->flags = 0;
-
 	r = dm_btree_empty(&pmd->info, &pmd->root);
 	if (r < 0)
 		goto bad_data_sm;
